server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -298,8 +298,10 @@ public void onResponse(String newGeneration) {
 
                     @Override
                     public void onFailure(Exception e) {
+                        final String failure = ExceptionsHelper.stackTrace(e);
```

Comment:
Ew. Can we follow up with a change that keeps the exception as an exception rather than converting it to a `String` here and in a few other places? Looks nontrivial because BWC, of course.

Reply:
Jup, I'm happy to try :)